src/DynamoPackages/PackageLoader.cs

```diff
@@ -399,6 +440,78 @@ public Package ScanPackageDirectory(string directory)
             return null;
         }
 
+        private static void CheckPackageNodeLibraryCertificates(string directory, Package discoveredPkg)
+        {
+            var dllfiles = new System.IO.DirectoryInfo(discoveredPkg.BinaryDirectory).EnumerateFiles("*.dll");
+            if (discoveredPkg.Header.node_libraries.Count() == 0 && dllfiles.Count() != 0)
+            {
+                throw new LibraryLoadFailedException(directory,
+                    String.Format(
+                        "A package called {0} found at {1} includes dll files but none are defined in node libraries in the package manifest.  Ignoring it.",
+                        discoveredPkg.Name, discoveredPkg.RootDirectory));
+            }
+
+            foreach (var manifestFile in discoveredPkg.Header.node_libraries)
+            {
+
+                //Try to get the assembly name from the manifest file
+                string filename;
+                try
+                {
+                    filename = new AssemblyName(manifestFile).Name + ".dll";
+                }
+                catch
+                {
+                    throw new LibraryLoadFailedException(directory,
+                        String.Format(
+                            "A package called {0} found at {1} has improperly defined its node libraries in the package manifest.  Ignoring it.",
+                            discoveredPkg.Name, discoveredPkg.RootDirectory));
+                }
+
+
+                //Verify the node library exists in the package bin directory
+                var filepath = Path.Combine(discoveredPkg.BinaryDirectory, filename + ".dll");
+                if (!File.Exists(filepath))
+                {
+                    throw new LibraryLoadFailedException(directory,
+                        String.Format(
+                            "A package called {0} found at {1} is missing dlls which are defined in the package manifest.  Ignoring it.",
+                            discoveredPkg.Name, discoveredPkg.RootDirectory));
+                }
+
+                //Verify that you can load the node library assembly into a Reflection only context
+                Assembly asm;
+                try
+                {
+                    asm = Assembly.ReflectionOnlyLoadFrom(filename);
```

Comment:
I really hope this doesn't cause issues with loading the assembly later if it's already loaded here?

Reply:
can we verify this somehow?